src/routers/mailingList/mailingList.ts

```diff
@@ -0,0 +1,39 @@
+import express, { Request, Response } from 'express';
+import { GoogleSpreadsheet } from 'google-spreadsheet';
+import { isMailingListRequestBody } from './utils';
+import { DEFAULT_EVENT_RESPONSES } from './constant';
+import GoogleCreds from '../../config/acs-web-5aaaf-768e669e2d39.json';
+
+const doc = new GoogleSpreadsheet(
+  '16PEbh4LFsb1klDX-W6kp47_p9acEKVIWNBz-TTDXcDc',
+);
+
+const router = express.Router();
+
+// add to mailing list with responses of interest in events and other feedback
+router.post('/api/mailingList', async (req: Request, res: Response) => {
+  if (!isMailingListRequestBody(req.body)) {
```

Comment:
I think this is great initiative to check the body!💯

But here's where I kind of want to follow a practice for all routers...
I prefer to have all routers wrapped in a `try catch`. 

Instead what you could have done was something like:
```
router.post('/api/mailingList', async (req: Request, res: Response) => {
  try {
    const { email, interestedEvents, otherFeedback } = req.body;
    if(!email || !interestedEvents || !otherFeedback) throw new Error(M.BAD_BODY);
    ...
  } catch (err) {
    console.log(err);
    res.status(400).send(error.message);
  }
});
```

It's kind of similar to what you did, however, it's a little more readable because the route will only have two `res.send()` expressions. One if the request was successful, and one if the request was not.

You could imagine that for bigger routes there might be multiple points of failure and it's super convenient to just 
`throw new Error('Message saying what went wrong');`

What do you think?

Reply:
That makes a lot of sense! We can also specify the HTTP status code in the error that we throw.

Reply:
Yoo I forgot about this but we can define our own error handler middleware and we don't 
even have to catch in the routes 😮 
https://expressjs.com/en/guide/error-handling.html

Reply:
Or rather we might still have to catch unfortunately but we just have to send along the error with `next(err)` and all routes can share the error handler